src/MongoDB.Analyzer/Core/Builders/BuilderExpressionProcessor.cs

```diff
@@ -143,11 +143,32 @@ private static IEnumerable<SyntaxNode> GetBuildersDefinitionNodes(SemanticModel
             {
                 nodesProcessed.Add(node);
 
-                // Skip MongoDB.Driver.Builders<T> nodes
+                var isBuildersContainer = false;
+                var containsAlias = false;
+
+                //Check if Member Expression is Builders Container
                 if (node is MemberAccessExpressionSyntax memberAccessExpressionSyntax &&
                     semanticModel.GetSymbolInfo(memberAccessExpressionSyntax.Expression).Symbol is INamedTypeSymbol namedTypeSymbol &&
-                    namedTypeSymbol.IsBuildersContainer() &&
-                    semanticModel.GetAliasInfo(memberAccessExpressionSyntax.Expression) == null)
+                    namedTypeSymbol.IsBuildersContainer())
+                {
+                    isBuildersContainer = true;
+                }
+
+                //Check if Node contains alias
+                var syntaxNode = (node as MemberAccessExpressionSyntax)?.Expression;
+                while (syntaxNode != null)
+                {
+                    if (semanticModel.GetAliasInfo(syntaxNode) != null)
+                    {
+                        containsAlias = true;
+                        break;
+                    }
+
+                    syntaxNode = (syntaxNode as MemberAccessExpressionSyntax)?.Expression;
+                }
+
+                //Skip Nodes that are Builder Containers and that don't contain alias
```

Comment:
Add two examples in the comment:
// For example MongoDB.Driver.Builders<T>, driverAlias.Builders<T>

Reply:
Done